sources/api/apiclient/src/main.rs

```diff
@@ -1486,4 +1568,91 @@ mod tests {
             _ => panic!("Expected Get with Prefixes"),
         }
     }
+
+    // Helper macro for creating network subcommands - extensible for future network subcommands
+    macro_rules! network_cmd {
+        // For Network::Configure subcommand with input_source
+        (configure, $input_source:expr) => {
+            Subcommand::Network(NetworkSubcommand::Configure(NetworkConfigureArgs {
+                input_source: $input_source.to_string(),
+            }))
+        }; // Future network subcommands can be added here, e.g.:
+           // (status) => { Subcommand::Network(NetworkSubcommand::Status(NetworkStatusArgs {})) };
+    }
+
+    #[test_case("apiclient network configure file:///tmp/net.toml",
+        global_args!(LevelFilter::Info),
+        network_cmd!(configure, "file:///tmp/net.toml");
+        "network configure with file URI")]
+    #[test_case("apiclient network configure base64:dmVyc2lvbiA9IDIKCltldGgwXQpkaGNwNCA9IHRydWU=",
+        global_args!(LevelFilter::Info),
+        network_cmd!(configure, "base64:dmVyc2lvbiA9IDIKCltldGgwXQpkaGNwNCA9IHRydWU=");
+        "network configure with base64 URI")]
+    fn test_network_parsing(cmd_str: &str, expected_args: Args, expected_subcommand: Subcommand) {
+        // Given a command line string for network configure
+        let (global_args, subcommand) = parse_command_line(cmd_str);
+
+        // When parsing the command line
+        // Then the global arguments should match what we expect
+        assert_eq!(global_args.log_level, expected_args.log_level);
+        assert_eq!(global_args.socket_path, expected_args.socket_path);
+
+        // And the subcommand should match what we expect
+        match (&subcommand, &expected_subcommand) {
+            (
+                Subcommand::Network(NetworkSubcommand::Configure(actual)),
+                Subcommand::Network(NetworkSubcommand::Configure(expected)),
+            ) => {
+                assert_eq!(actual.input_source, expected.input_source);
+            }
+            _ => panic!(
+                "Expected Network::Configure subcommand: {expected_subcommand:?}, got: {subcommand:?}"
+            ),
+        }
```

Comment:
The macro above is pretty effortful in making it easy to test new subcommands, but then we have to maintain this match tree.

Can we just `#[derive(PartialEq)]` on `Subcommand`?

Reply:
Done. I had to add `#derive(PartialEq)` to all the relevant structs referred by `Subcommand`.